x-pack/legacy/plugins/epm/server/lib/field.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// This should become a copy of https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/mapping/field.go#L39
+export interface Field {
+  name: string;
+  type: string;
+  required: boolean;
+  description: string;
+  fields: Field[];
+}
+
+type Fields = Field[];
+
+/**
```

Comment:
One important thing to note is that this function mutates the data structure it's given. It's generally considered poor form to mutate arguments directly, but there are times when it's preferable/required. 

I think it'd help to add a note here that this function alters the structures it's given in-place. Perhaps even a different name, but I don't have suggestions. You get some signal when you see `ProcessFields(fields)` with no return value, but like I said it's uncommon so anything we can do to highlight the behavior is appreciated.